ViewModel/BrowserViewModel.swift

```diff
@@ -391,4 +392,34 @@ class BrowserViewModel: NSObject, ObservableObject,
             }
         }
     }
+
+    // MARK: - New tab in macos desktop
+      
+      #if os(macOS)
+      func webView(_ webView: WKWebView, createWebViewWith configuration: WKWebViewConfiguration,
+                   for navigationAction: WKNavigationAction, windowFeatures: WKWindowFeatures) -> WKWebView? {
+          
+          guard navigationAction.targetFrame == nil else { return nil }
+          guard let newUrl = navigationAction.request.url else { return nil }
+          
+          // open external link in default browser
+          if newUrl.isExternalLink {
+              self.openExternalLinkURL = newUrl
+              return nil
+          }
+
+          // create new tab
+          guard let currentWindow = NSApp.keyWindow,
+                let windowController = currentWindow.windowController else { return nil }
+          windowController.newWindowForTab(nil)
+          guard let newWindow = NSApp.keyWindow, currentWindow != newWindow else { return nil }
+          currentWindow.addTabbedWindow(newWindow, ordered: .above)
+          
+          // open url in new tab
+          NotificationCenter.openURL(newUrl, inNewTab: true)
```

Comment:
```suggestion
          NotificationCenter.openURL(newUrl)
```
Context: `inNewTab` is for iOS / iPadOS only, it doesn't do anything for Mac, so I suggest removing it